server/service/mdm.go

```diff
@@ -1120,8 +1120,10 @@ func (svc *Service) DeleteMDMWindowsConfigProfile(ctx context.Context, profileUU
 		return ctxerr.Wrap(ctx, err)
 	}
 
-	// TODO: integrate the call to bulk-update host profiles affected by this deletion (see Apple's implementation)
-	// (part of https://github.com/fleetdm/fleet/issues/14364)
+	// cannot use the profile ID as it is now deleted
+	if err := svc.ds.BulkSetPendingMDMHostProfiles(ctx, nil, []uint{teamID}, nil, nil, nil); err != nil {
```

Comment:
I have a vague memory of a case where we were doing something like that and resulted in performance issues, does that ring a bell to you? I think it's a scenario where we tested with lots of teams... Just want to make sure we're not reintroducing a similar performance problem here, because I think it was related to a deletion like that where we were falling back on a more general/wider-reach id.

Reply:
Looks like this is how we handled the Apple profile deletion, so that's probably fine and I was thinking of something else!